src/crypto/hsm/infineon/CHIPCryptoPALHsm_P256_trustm.cpp

```diff
@@ -59,93 +59,92 @@ CHIP_ERROR P256KeypairHSM::Initialize(ECPKeyTarget key_target)
     {
         // Trust M ECC 256 Key Gen
         ChipLogDetail(Crypto, "Generating NIST256 key in Trust M !");
-        uint8_t key_usgae = (optiga_key_usage_t)( OPTIGA_KEY_USAGE_KEY_AGREEMENT | OPTIGA_KEY_USAGE_SIGN | OPTIGA_KEY_USAGE_AUTHENTICATION );
-        return_status = trustm_ecc_keygen(OPTIGA_KEY_ID_E0F2, key_usgae, OPTIGA_ECC_CURVE_NIST_P_256, pubkey, pubKeyLen); 
+        uint8_t key_usgae =
+            (optiga_key_usage_t)(OPTIGA_KEY_USAGE_KEY_AGREEMENT | OPTIGA_KEY_USAGE_SIGN | OPTIGA_KEY_USAGE_AUTHENTICATION);
+        return_status = trustm_ecc_keygen(OPTIGA_KEY_ID_E0F2, key_usgae, OPTIGA_ECC_CURVE_NIST_P_256, pubkey, pubKeyLen);
         VerifyOrExit(return_status == OPTIGA_LIB_SUCCESS, error = CHIP_ERROR_INTERNAL);
     }
     else
     {
-        //Read out the public Key stored
+        // Read out the public Key stored
         ChipLogDetail(Crypto, "Provisioned_key - %lx !", keyid);
-        trustmGetKey(TRUSTM_P256_PUBKEY_OID_KEY,pubkey,&pubKeyLen); 
+        trustmGetKey(TRUSTM_P256_PUBKEY_OID_KEY, pubkey, &pubKeyLen);
 
         /* Set the public key */
         P256PublicKeyHSM & public_key = const_cast<P256PublicKeyHSM &>(Pubkey());
-        VerifyOrReturnError((size_t)pubKeyLen > NIST256_HEADER_OFFSET, CHIP_ERROR_INTERNAL);
-        VerifyOrReturnError(((size_t)pubKeyLen - NIST256_HEADER_OFFSET) <= kP256_PublicKey_Length, CHIP_ERROR_INTERNAL);
+        VerifyOrReturnError((size_t) pubKeyLen > NIST256_HEADER_OFFSET, CHIP_ERROR_INTERNAL);
+        VerifyOrReturnError(((size_t) pubKeyLen - NIST256_HEADER_OFFSET) <= kP256_PublicKey_Length, CHIP_ERROR_INTERNAL);
         memcpy((void *) Uint8::to_const_uchar(public_key), pubkey, pubKeyLen);
         public_key.SetPublicKeyId(keyid);
-    
-    } 
+    }
     error = CHIP_NO_ERROR;
-    exit:
-        if (error != CHIP_NO_ERROR)
-        {
-            trustm_close();
-        }
-        return error;
+exit:
+    if (error != CHIP_NO_ERROR)
+    {
+        trustm_close();
```

Comment:
Should `trustm_close()` not always be called on success as well?

Reply:
Hi @tcarmelveilleux, Thanks for your comments. trustm_close() is not always needed to be called on success in our implementation.